components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/jms/JMSPollingConsumer.java

```diff
@@ -228,6 +255,27 @@ public Message poll() {
         } catch (Exception e) {
             logger.error("Error while receiving JMS message. " + e.getMessage(), e);
         } finally {
+            if (!isConnected) {
+                if (reconnectDuration != null) {
+                    retryDuration = reconnectDuration;
+                    logger.error("Reconnection attempt : " + (retryIteration++) + " for JMS Inbound : " +
+                            name + " failed. Next retry in " + (retryDuration / 1000) +
+                            " seconds. (Fixed Interval)");
+                } else {
+                    retryDuration = (long) (retryDuration * reconnectionProgressionFactor);
+                    if (retryDuration > maxReconnectDuration) {
+                        retryDuration = maxReconnectDuration;
+                        logger.info("InitialReconnectDuration reached to MaxReconnectDuration.");
+                    }
+                    logger.error("Reconnection attempt : " + (retryIteration++) + " for JMS Inbound : " +
+                            name + " failed. Next retry in " + (retryDuration / 1000) +
+                            " seconds");
+                }
+                try {
+                    Thread.sleep(retryDuration);
+                } catch (InterruptedException ignore) {
+                }
```

Comment:
Better if we can add a comment here mentioning, why we are specifically ignoring this exception